docs/incubator-proposal.md

```diff
@@ -0,0 +1,194 @@
+# cert-manager proposal
+
+## Problem
+
+Currently in Kubernetes, a number of different components including user
+applications require TLS certificates. Right now, users can provide
+certificates to workloads on the cluster by creating `Secret` resources that
+contain a private/public keypair.
+
+A number of projects (see: [kube-lego](https://github.com/jetstack/kube-lego),
+[kube-cert-manager](https://github.com/PalmStoneGames/kube-cert-manager)
+and [kubernetes-letsencrypt](https://github.com/tazjin/kubernetes-letsencrypt))
+came about in order to automate the retrieval of certificates from an [ACME](https://github.com/ietf-wg-acme/acme/)
+compliant server. This has served the community well, especially with the rise
+in popularity of ACME due to [letsencrypt](https://letsencrypt.org). These
+projects also attempt to renew the certificate as it reaches it's expiry time.
+
+Whilst each project has invented it's own schema/methodology for obtaining
+these certificates, the end goal has largely been the same. To provide a signed
```

Comment:
> end goal

Are these certificates (publicly trusted) (server auth) certificates? Do you want to extend these to private CAs and other usages?

Reply:
It's not the concern of cert-manager who trusts the certificates - right now it is unable to determine whether a pre-existing certificate is signed by the issuing authority, but in future I'd like to add a `CA` method to the issuer interface so that actual verification of the Certificate can be performed (at which point a re-issue could be performed).